### PR TITLE
add support for custom settings in calltaker

### PR DESCRIPTION
**Pull request by daniel-heppner-ibigroup**
**Description:**
This PR adds support for custom mode settings in the calltaker advanced options panel. 

**PR Checklist:**
- [ ] Does the code follow accessibility standards (WCAG 2.1 AA Compliant)?
- [ ] Are all languages supported (Internationalization/Localization)?
- [ ] Are appropriate Typescript types implemented?

image

### CODE REVIEW

**Comment by daniel-heppner-ibigroup** [comment]
Currently relies on an alpha build

**Comment by daniel-heppner-ibigroup** [comment]
Actually, I'm going to have @miles-grant-ibigroup review this first because he did some work on these files recently, and I removed some code he wrote.

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
Code looks great, nice and clean! Only issue is I don't see how to hide walk reluctance on bike+transit or bikesahre+transit. what do I put in the config? Is this possible?

Also is it possible to go back to the old layout? There's almost no room for results now on high zoom levels/low screen resolutions

**Comment by daniel-heppner-ibigroup** [comment]
Hm, the layout thing is a problem. I'm not sure how the old layout would work given that there's the ability to have multiple settings. Isn't it fine if the results aren't visible while advanced options are open? 

Otherwise, the reason that walk reluctance is visible is because it's associated with the "transit" mode. We need to add "walk" to the transit only option, then assign walk reluctance mode setting to "walk" instead of "transit". This is the same as the regular mode selector.

**Comment by daniel-heppner-ibigroup** [comment]
OK, so I made it show the first one in the old position and any extras are displayed below.

**Review by miles-grant-ibigroup — APPROVED** [review]
Great stuff!

**Comment by daniel-heppner-ibigroup** [comment]
I updated the version of trip-details in this PR to get the new fare changes.

File: lib/components/form/call-taker/advanced-options.js [hunk 1]
```diff
@@ -234,33 +223,14 @@ class AdvancedOptions extends Component {
...
-              bikeEnabled
-                ? this.props.modeSettingValues.bikeReluctance
-                : this.props.modeSettingValues.walkReluctance
-            }
-          />
-
+          {/* Show the first mode setting */}
+          {applicableModeSettings.length >= 1 && (
+            <ModeSettingRenderer
+              key={applicableModeSettings[0].key}
```
Comment (binh-dam-ibigroup) [inline lib/components/form/call-taker/advanced-options.js:229]: What is the need for the `key` prop here?

**Review by binh-dam-ibigroup — APPROVED** [review]
LGTM

**Pull request merged**